Spec/RealtimeClientConnection.swift

```diff
@@ -3320,6 +3320,52 @@ class RealtimeClientConnection: QuickSpec {
                     }
                 }
 
+                // RTN22
+                it("Ably can request that a connected client re-authenticates by sending the client an AUTH ProtocolMessage") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    options.useTokenAuth = true
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    client.setTransportClass(TestProxyTransport.self)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.Connected) { stateChange in
+                            expect(stateChange?.reason).to(beNil())
+                            done()
+                        }
+                        client.connect()
+                    }
+
+                    guard let initialToken = client.auth.tokenDetails?.token else {
+                        fail("Initial token is nil"); return
+                    }
+
+                    guard let transport = client.transport as? TestProxyTransport else {
+                        fail("TestProxyTransport is not set"); return
+                    }
+
+                    var authorizeMethodCallCount = 0
+                    let hook = client.auth.testSuite_injectIntoMethodAfter(#selector(client.auth.authorize)) {
+                        authorizeMethodCallCount += 1
+                    }
+                    defer { hook.remove() }
+
+                    let authMessage = ARTProtocolMessage()
+                    authMessage.action = .Auth
+                    transport.receive(authMessage)
```

Comment:
We may in fact not need to stub which makes the tests more resilient.  Why not issue a token with a `ttl` of 35 seconds, and within 5 seconds you will receive the `AUTH` protocol message naturally.  That way all of this stubbing becomes unnecessary